### PR TITLE
3 nearby screen

### LINKED ISSUE #3: Nearby screen shows nearby bus stops
We need a view that shows nearby bus stops on a Google Maps view.
- tapping on a marker should bring up stop information and suggest the user add it to favourites

### PR DESCRIPTION
**Pull request by yenuGH**
### Added Features:
- Realtime location tracking updates dispatched into background IO thread
    - Green marker to indicate user's current location
    - Red markers dynamically go in and out of scope when in range of blue radius circle
- BottomSheet added with recycler view to show cards of nearby bus stops with more information
    - Has buttons to save and discard (will later be used for saving to favourites or adding to a trip)
    - Each card can be expanded for additional information for a specific bus stop
- Clicking on any marker will disable centering of user's location until the recenter button is pressed
    - Recenter button icon changes dynamically to reflect whether location tracking is on or off

### Todo:
- Make more efficient algorithms to dynamically refresh markers and bus stops
- See if location tracking animations are able to become smoother
- Allow for tapping on a marker to open up the BottomSheet expanding the corresponding bus stop's info card
- Integrate loading data from mongo database rather than csv file
- make prettier c:

### CODE REVIEW

File: app/src/main/java/ca/wheresthebus/data/model/Stop.kt
Comment (toastjpg) [inline app/src/main/java/ca/wheresthebus/data/model/Stop.kt:1]: delete this

File: app/build.gradle.kts [hunk 1]
```diff
@@ -35,6 +35,9 @@ android {
         versionName = "1.0"
 
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+
+        // this build config field is to set the maps api key in the manifest properly - from chatgpt :)
+        buildConfigField("String", "MAPS_KEY", "\"${project.findProperty("MAPS_KEY")}\"");
```
Comment (bryanjhdang) [inline app/build.gradle.kts:40]: I'm not sure if you need this anymore as the build config is generated from values `secrets.properties` but it's fine for now, can take a look into it in the future

Comment (yenuGH) [reply app/build.gradle.kts:40]: i tried taking this out my google maps still refused to load without this. maybe it's something wrong with my end but not sure what's causing it

**Comment by bryanjhdang** [comment]
LGTM

**Pull request merged**